library/src/main/java/com/firebase/ui/auth/core/AuthProviderType.java

```diff
@@ -0,0 +1,61 @@
+package com.firebase.ui.auth.core;
+
+import android.content.Context;
+
+import com.firebase.client.Firebase;
+import com.firebase.ui.R;
+import com.firebase.ui.auth.facebook.FacebookAuthProvider;
+import com.firebase.ui.auth.google.GoogleAuthProvider;
+import com.firebase.ui.auth.password.PasswordAuthProvider;
+import com.firebase.ui.auth.twitter.TwitterAuthProvider;
+
+import java.lang.reflect.InvocationTargetException;
+
+public enum AuthProviderType {
+    GOOGLE  ("google",   "com.firebase.ui.auth.google.GoogleAuthProvider",     R.id.google_button),
```

Comment:
Can we not do just "google.GoogleAuthProvider" and assume the "com.firebase.ui.auth" in the constructor? I feel like that might read better